test/any/datastore/updater-integration.es6.js

```diff
@@ -0,0 +1,182 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+describe('Datastore updater (set ops + versioned DAOs)', () => {
+  var MDAO;
+  var QuickSink;
+  var SyncDAO;
+  var VersionedSyncRecord;
+  var VersionNoDAO;
+  var DatastoreUpdater;
+  var Item;
+  var VersionedItem;
+  var srcDAO;
+  var cacheDAO;
+  var dstDAO;
+  var datastoreDAO;
+  var updater;
+  beforeEach(() => {
+    MDAO = foam.lookup('foam.dao.MDAO');
+    QuickSink = foam.lookup('foam.dao.QuickSink');
+    SyncDAO = foam.lookup('foam.dao.SyncDAO');
+    VersionedSyncRecord = foam.lookup('foam.dao.sync.VersionedSyncRecord');
+    VersionNoDAO = foam.lookup('foam.dao.VersionNoDAO');
+    DatastoreUpdater = foam.lookup('org.chromium.apis.web.DatastoreUpdater');
+
+    foam.CLASS({
+      package: 'org.chromium.mlang.sink.test',
+      name: 'Item',
+
+      properties: [
+        {
+          class: 'Int',
+          name: 'id',
+        },
+        {
+          name: 'data',
+        },
+      ],
+    });
+
+    Item = foam.lookup('org.chromium.mlang.sink.test.Item');
+    VersionedItem = foam.lookup('foam.version.VersionedClassFactorySingleton')
+        .create().get(Item);
+    srcDAO = MDAO.create({of: Item});
+    cacheDAO = MDAO.create({of: Item});
+    datastoreDAO = MDAO.create({of: VersionedItem});
+    dstDAO = SyncDAO.create({
+      of: VersionedItem,
+      delegate: MDAO.create({of: VersionedItem}),
+      syncRecordDAO: MDAO.create({of: VersionedSyncRecord}),
+      remoteDAO: VersionNoDAO.create({
+        delegate: datastoreDAO,
+      }),
+      polling: true,
+    });
+    updater = DatastoreUpdater.create();
+  });
+
+  it('should only perform necessary put/remove updates', done => {
```

Comment:
What is `it` and why does it have a callback instead of a promise?

Reply:
`it` is JasmineJS's test spec API. It may have a promisey API by now, but I'm used to depending on a version of JasmineJS that uses "you specify a single 'done' argument" as the API for async tests.